pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -125,6 +125,8 @@ def type(self, dst_type: Union[str, torch.dtype]) -> torch.nn.Module:
         Returns:
             Module: self
         """
+        if isinstance(dst_type, str):
```

Comment:
not sure this is working for all the types. Even if it is. Can't we maybe find a better way to do this? For devices, you can simply do `device = torch.device(device_str)`. I know, this does not work for dtypes, but maybe something similar? But in general: why doesn't strings work with torchscript?